frontend/src/metabase/query_builder/components/view/ViewFooter.jsx

```diff
@@ -142,19 +144,23 @@ const ViewFooter = ({
               }
             />
           ),
-          type === "question" && (
-            <EmbedMenu
-              key="embed"
-              resource={question}
-              resourceType="question"
-              hasPublicLink={!!question.publicUUID()}
-              onModalOpen={() =>
-                question.isSaved()
-                  ? onOpenModal("embed")
-                  : onOpenModal("save-question-before-embed")
-              }
-            />
-          ),
+          type === "question" &&
```

Comment:
I moved the `question.isSaved()` logic up here. It almost mimics `DashboardEmbedAction` where it gatekeeps the actual menu until the question is saved.

Reply:
I like this. It's just that now we're using `DashboardEmbedHeaderButton` inside `query_builder/` directory which is for questions.

If this isn't specifically for dashboard anymore we should rename it accordingly. Like to just `ResourceEmbedHeaderButton` or something.

Reply:
Will change right now. Good call

Reply:
@WiNloSt let me know what you think. I renamed `DashboardHeaderEmbedButton` to `ResourceEmbedButton` (in dashboards, the button is in the header, but in questions it's in the footer so "Header" is not the greatest name).

Also I moved the component into the `public/components` folder, together with the `EmbedModal` and `EmbedFrame` components which the button leads to in the user flow.